### ISSUE TITLE
[BUG] The pagination is not reset on a filter change

### ISSUE THREAD

**Issue by TheFehr** [avo-hq/avo#428]
### Describe the bug
On a Resource Index with a select filter, with a select filter active (an option selected) and you navigate to later pages and then change the filter to a selection that has less pages, it throws an error.

### Steps to Reproduce
1. Go to a resource index page
2. Select a filter option
3. Change the page
4. Select another filter option that results in less pages

### Expected behavior
Either navigate to the last or the first page of the filtered view

### Actual behavior
Pagy::OverflowError Exception

### System configuration
**Avo version**:
1.7.2

**Rails version**:
6.0.3.6

**Ruby version**:
2.7.3p183

**Comment by TheFehr** [comment]
I could share a sentry.io issues covering that bug if that helps. Feel free to contact me if that is of interest.

**Comment by adrianthedev** [comment]
Hmm. Yes. I understand where the bug is. 

Basically if the user changes the filters, the pagination should reset automatically. 

Thanks for submitting.

**Comment by adrianthedev** [comment]
We'll release a bugfix release this coming week.

**Comment by adrianthedev** [comment]
Released in [1.7.3](https://rubygems.org/gems/avo/versions/1.7.3)

**Comment by stefsava** [comment]
Seem yet present
for example if from

https://main.avodemo.com/avo/resources/projects?keep_filters_panel_open&page=2

filter by a country
https://main.avodemo.com/avo/resources/projects?page=2&filters[country][is][]=GT
I receive a error 500 on main.avodemo.com
or 
Pagy::OverflowError at /resources/ea_service_categories
expected :page in 1..1; got 2
in local env

**Comment by adrianthedev** [comment]
Thank you @stefsava.

I created a new issue with this and added the `Help wanted` label. Maybe someone can help before we can reach out to fix it.
